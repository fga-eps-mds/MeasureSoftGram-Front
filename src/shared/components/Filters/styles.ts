import { Box } from '@mui/material';
import styled from 'styled-components';

export const FiltersTitle = styled.button`
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: #ffffff;
  margin: 0.4rem;

  outline: 0;
  border: 0;
  transition: 0.2s all;
  cursor: pointer;

  &:hover {
    opacity: 0.6;
    border-radius: 2px;
    background-color: #f1f1f1;
  }

  span {
    font-weight: 700;
    font-size: 1rem;
    line-height: 1.5rem;
    color: #000000;

    padding: 0.4rem;
  }
`;

export const OptionContainer = styled(Box)``;

export const Option = styled(Box)`
  & > span:first-child {
    padding: 0.2rem;
    margin-left: 1rem;
  }

  p {
    font-weight: 400;
    font-size: 0.8rem;
    line-height: 1rem;
  }
`;
