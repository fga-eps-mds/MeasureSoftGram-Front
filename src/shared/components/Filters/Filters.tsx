/* eslint-disable react/no-array-index-key */
import React, { useState } from 'react';
import { ExpandCircleDown, ExpandCircleDownOutlined } from '@mui/icons-material';
import { Checkbox } from '@mui/material';
import undelineRemover from '@utils/undelineRemover';
import capitilizer from '@utils/capitilizer';
import { FiltersTitle, Option, OptionContainer } from './styles';

interface OptionCheckedProps {
  [key: string]: boolean;
}

export interface FiltersProps {
  filterTitle: string;
  options?: Array<string>;
  updateOptions: Function;
  checkedOptions: OptionCheckedProps;
}

const Filters: React.FC<FiltersProps> = ({ filterTitle, options = [], updateOptions, checkedOptions }) => {
  const [isVisible, setIsVisible] = useState<boolean>(true);

  const handleUpdate = (selectedOption: string) => {
    updateOptions({
      ...checkedOptions,
      [selectedOption]: !checkedOptions[selectedOption]
    });
  };

  return (
    <div>
      <FiltersTitle onClick={() => setIsVisible(!isVisible)}>
        {isVisible ? <ExpandCircleDown /> : <ExpandCircleDownOutlined />}
        <span>{filterTitle}</span>
      </FiltersTitle>

      {isVisible ? (
        <OptionContainer display="flex" flexDirection="column">
          {options.map((option) => (
            <Option
              onClick={() => handleUpdate(option)}
              data-testid={option}
              key={option}
              display="flex"
              flexDirection="row"
              alignItems="center"
              sx={{ cursor: 'pointer' }}
            >
              <Checkbox checked={checkedOptions[option]} />
              <span>{capitilizer(undelineRemover(option))}</span>
            </Option>
          ))}
        </OptionContainer>
      ) : null}
    </div>
  );
};

export default Filters;
