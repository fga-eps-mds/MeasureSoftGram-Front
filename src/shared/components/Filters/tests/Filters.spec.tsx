import { act, fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';

import Filters from '../Filters';

const FILTER_TITLE = 'FILTER_TITLE';
const OPTIONS = ['Op_1', 'Op_2', 'Op_3'];
const CHECKED_OPTIONS = {
  Op_1: false,
  Op_2: false,
  Op_3: true
};

const defaultProps = {
  checkedOptions: CHECKED_OPTIONS,
  updateOptions: jest.fn(),
  filterTitle: FILTER_TITLE,
  options: OPTIONS
};

type defaultPropsType = typeof defaultProps;

const renderFilter = (props: defaultPropsType) => render(<Filters {...props} />);

describe('<Filter />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Snapshot', () => {
    it('Deve corresponder ao Snapshot', () => {
      const tree = renderFilter(defaultProps);
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Comportamento', () => {
    it('Deve chamar updateOptions ao selecionar novo checkbox', () => {
      const updateOptions = jest.fn();
      const register = renderFilter({ ...defaultProps, updateOptions });

      const checkbox = register.queryByTestId(OPTIONS[0]);
      act(() => {
        fireEvent.click(checkbox!, { target: { checked: true } });
      });
      expect(updateOptions).toBeCalled();
    });
  });
});
