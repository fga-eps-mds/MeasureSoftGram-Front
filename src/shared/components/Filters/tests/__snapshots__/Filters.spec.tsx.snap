// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="styles__FiltersTitle-sc-1ywle5j-0 ePgPyn"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandCircleDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
            />
          </svg>
          <span>
            FILTER_TITLE
          </span>
        </button>
        <div
          class="styles__OptionContainer-sc-1ywle5j-1 MuiBox-root css-j7qwjs"
        >
          <div
            class="styles__Option-sc-1ywle5j-2 kZhFnX MuiBox-root css-1qx36yh"
            data-testid="Op_1"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span>
              Op 1
            </span>
          </div>
          <div
            class="styles__Option-sc-1ywle5j-2 kZhFnX MuiBox-root css-1qx36yh"
            data-testid="Op_2"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span>
              Op 2
            </span>
          </div>
          <div
            class="styles__Option-sc-1ywle5j-2 kZhFnX MuiBox-root css-1qx36yh"
            data-testid="Op_3"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span>
              Op 3
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="styles__FiltersTitle-sc-1ywle5j-0 ePgPyn"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandCircleDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
          />
        </svg>
        <span>
          FILTER_TITLE
        </span>
      </button>
      <div
        class="styles__OptionContainer-sc-1ywle5j-1 MuiBox-root css-j7qwjs"
      >
        <div
          class="styles__Option-sc-1ywle5j-2 kZhFnX MuiBox-root css-1qx36yh"
          data-testid="Op_1"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span>
            Op 1
          </span>
        </div>
        <div
          class="styles__Option-sc-1ywle5j-2 kZhFnX MuiBox-root css-1qx36yh"
          data-testid="Op_2"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span>
            Op 2
          </span>
        </div>
        <div
          class="styles__Option-sc-1ywle5j-2 kZhFnX MuiBox-root css-1qx36yh"
          data-testid="Op_3"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span>
            Op 3
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
